scipy/signal/_short_time_fft.py

```diff
@@ -0,0 +1,1659 @@
+"""Short-time Fourier Transform Module.
+
+Implementation Notes (as of 2023-04)
+------------------------------------
+* MyPy version 1.1.1 does not seem to support decorated property methods
+  properly. Hence, applying ``@property`` to methods decorated with `@cache``
+  (as tried with the ``lower_border_end`` method) causes a mypy error when
+  accessing it as an index (e.g., ``SFT.lower_border_end[0]``).
+* MyPy version 1.1.1 does not seem to support ``NDArray[complex]`` (only
+  ``NDArray[np.complex128]``). Hence the methods `stft()`, `stft_detrend()`,
+  `istft()`, and  `_ifft_func()` need to use `NDArray` array as a parameter
+  instead of `NDArray[complex]`.
+* The entry ``.. currentmodule:: scipy.signal.ShortTimeFFT`` is required
+  in docstrings to ensure that `stft` and `istft` generate HTML links to
+  methods and not the legacy functions. The side effect is that `ShortTimeFFT`
+  does not work anymore. It is not known how to do this file-wide instead of
+  method-wide.
+* The HTML documentation currently renders each method/property on a separate
+  page without reference to the parent class. Thus, a link to `ShortTimeFFT`
+  was added to the "See Also" section of each method/property.
```

Comment:
This should not be needed, see other examples of classes for how we document these usually. e.g. `scipy.stats.qmc.Sobol`.

Reply:
On the page [scipy.stats.qmc.Sobol.random](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.Sobol.random.html#scipy.stats.qmc.Sobol.random) I do not see a link to the parent class [scipy.stats.qmc.Sobol](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.Sobol.html). Such a link would simplify navigating around a bit, IMO.

Reply:
It might, but that's a larger issue and should not be dealt with here. It's more something to be done on the theme side itself.

That's a general thing, some choices and suggestions are made might not be "optimal", but we need to find a balance between "fixing" things and being consistent with the rest of the code base. Otherwise we end up with multiple ways of dealing with a problem and we are struggling then to maintain things.